cluster-autoscaler/README.md

```diff
@@ -1,4 +1,80 @@
-Cluster Autoscaler automatically adjusts the number of nodes in a cluster.
+# Cluster Autoscaler
 
-**Status:** under development
+# Introduction
 
+Cluster Autoscaler is a tool that automatically adjusts the size of the Kubernetes cluster when:
+* there is a pod that doesn’t have enough space to run in the cluster
+* some nodes in the cluster are so underutilized that they can be deleted and their pods will 
+be easily placed on some other, existing nodes.  
+
+# Deployment
+
+Cluster Autoscaler runs on the Kubernetes master node (at least in the default setup on GCE and GKE). 
+It is possible to run customized Cluster Autoscaler inside of the cluster but then extra care needs
+to be taken to ensure that Cluster Autoscaler is up and running. User can put it into kube-system
+namespace (Cluster Autoscaler doesn't scale down node with non-manifest based kube-system pods running
+on them) and mark with `scheduler.alpha.kubernetes.io/critical-pod` annotation (so that the rescheduler, 
+if enabled, will kill other pods to make space for it to run). 
+
+Right now it is possible to run Cluster Autoscaler on:
+* GCE http://kubernetes.io/docs/admin/cluster-management/#cluster-autoscaling
+* GKE https://cloud.google.com/container-engine/docs/cluster-autoscaler
+* AWS https://github.com/kubernetes/contrib/blob/master/cluster-autoscaler/cloudprovider/aws/README.md
+
+# Scale Up
+
+Scale up creates a watch on the api server looking for all pods. Every 10 seconds (configurable)
+it checks for any unschedulable pods. Unschedulable pods are recognized by their PodCondition. 
+Whenever a kubernetes scheduler fails to find a place to run a pod it sets "schedulable" 
+PodCondition to false and reason to "unschedulable".  If there are any items on the unschedulable 
+lists Cluster Autoscaler tries to find a new place to run them. 
+
+It is assumed that the underlying cluster is run on top of some kind of node groups.
+Inside a node group all machines have identical capacity and have the same set of assigned labels. 
+Thus increasing a size of a node pool will bring a couple of new machines that will be similar 
+to these that are already in the cluster - they will just not have the user-created pods (but 
+will have all pods run from the node manifest or daemon sets). 
+
+Based on the above assumption Cluster Autoscaler creates template nodes for each of the 
+node groups and checks if any of the unschedulable pods would fit to a brand new node, if created.
```

Comment:
I'd add that even though is similar to "scheduling" it is much easier than real scheduler and those decisions might not be optimal (e.g. we might need multiple iterations to actually schedule all pods)